### PR TITLE
[Devfile] Add a spinner for deployment rollout

### PR DESCRIPTION
**Pull request by johnmcollier**
Signed-off-by: John Collier <John.J.Collier@ibm.com>

**What type of PR is this?**
> Uncomment only one ` /kind` line, and delete the rest.
> For example, `> /kind bug` would simply become: `/kind bug`

/kind bug
/area devfile

**What does does this PR do / why we need it**:
This PR adds a spinner to the `WaitForDeploymentRollout` function in kclient. Previously, since there was no spinner, the user would be looking at

```
Johns-MacBook-Pro-3:springboot johncollier$ odo push
 •  Push devfile component java-spring-boot  ...
```

While they waited for the component to rollout. Now they will see:
```
Johns-MacBook-Pro-3:springboot johncollier$ odo push
 •  Push devfile component java-spring-boot  ...
 ◓  Waiting for component to start
```

and 

```
Johns-MacBook-Pro-3:springboot johncollier$ odo push
 •  Push devfile component java-spring-boot  ...
 ✓  Waiting for component to start [8s]
 ✓  Checking files for pushing [6ms]
 ✓  Syncing files to the component [2s]
 ✓  Executing devbuild command "/artifacts/bin/build-container-full.sh" [28s]
 ✓  Executing devrun command "/artifacts/bin/start-server.sh" [2s]
 ✓  Push devfile component java-spring-boot [42s]
 ✓  Changes successfully pushed to component
```

**Which issue(s) this PR fixes**:

Fixes https://github.com/openshift/odo/issues/2827

**How to test changes / Special notes to the reviewer**:
Run `odo push` with a devfile, and verify that the spinner shows up while the deployment rolls out

### CODE REVIEW

File: pkg/devfile/adapters/kubernetes/component/adapter.go [hunk 1]
```diff
@@ -86,7 +86,7 @@ func (a Adapter) Push(parameters common.PushParameters) (err error) {
 	}
 
 	// Wait for Pod to be in running state otherwise we can't sync data or exec commands to it.
-	pod, err := a.waitAndGetComponentPod(false)
+	pod, err := a.waitAndGetComponentPod(true)
```
Comment (GeekArthur) [inline pkg/devfile/adapters/kubernetes/component/adapter.go:89]: We set `hideSpinner` to true here, that means we don't need to spinner for waiting pods?

Comment (johnmcollier) [reply pkg/devfile/adapters/kubernetes/component/adapter.go:89]: The spinner is not necessary here, we're just using that to retrieve the pod. The actually waiting for the component to start is in the Rollout function.

**Comment by GeekArthur** [comment]
The changes look good to me, just leave a comment above
/lgtm

**Pull request merged**